File: app/src/main/java/com/example/patiobalmax/database/PatioDao.java
package com.example.patiobalmax.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface PatioDao {

    @Insert
    void insert(PatioEntity patio);

    @Update
    void update(PatioEntity patio);

    @Delete
    void delete(PatioEntity patio);

    @Query("SELECT * FROM patios")
    LiveData<List<PatioEntity>> getAll();
}

File: app/src/main/kotlin/com/example/patiobalmax/database/dao/PatioDao.kt
package com.example.patiobalmax.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.example.patiobalmax.database.entity.Patio

@Dao
interface PatioDao {

    @Insert
    suspend fun insert(patio: Patio)

    @Query("SELECT * FROM patios")
    suspend fun getAll(): List<Patio>

    @Query("SELECT * FROM patios WHERE id = :patioId")
    suspend fun getById(patioId: Int): Patio?
}
